extras/shell/src/main/java/org/apache/rya/shell/RyaConnectionCommands.java

```diff
@@ -199,6 +199,14 @@ public void run() {
                 }
             });
 
+            try {
+            	//attempt to get the connection point, essentially pinging mongo server.
+            	adminClient.getConnectPoint();
+            } catch (MongoException e) {
+            	//unable to reach the mongo server, throw expected failed exception.
+            	throw new IOException(e);
```

Comment:
Instead of rethrowing the exception, could you return with a message explaining what went wrong? Not connecting to a Mongo DB because it was not there isn't an exceptional state when you're in the shell. By returning early, the connection commands will still be presented to the user.

The message could be something like, "Could not connect to the specified Mongo DB."

You also need to close the adminClient that was just made before returning.